src/main/kotlin/link/infra/packwiz/installer/UpdateManager.kt

```diff
@@ -338,7 +339,7 @@ class UpdateManager internal constructor(private val opts: Options, val ui: IUse
 			if (!ui.optionsButtonPressed) {
 				// TODO: this is so ugly
 				ui.submitProgress(InstallProgress("Reconfigure optional mods?", 0,1))
-				ui.awaitOptionalButton(true)
+				ui.awaitOptionalButton(true, -1) // TODO: Should user interactivity be forced here?
```

Comment:
I couldn't figure out when this code happens, but since the lines above seem to prompt the user a question, I left this at `-1`/no timeout. If this is okay, I will amend the commit to remove the `TODO` comment.

Reply:
This occurs when a modpack has optional mods, and the user has previously installed the pack and selected optional mods (so the dialog to select optional mods hasn't opened, unless they press the optional mods button at this point), and an update is about to be installed. It should probably be the same as the other `awaitOptionalButton` calls.

Reply:
Ah, that explains why my simple test cases weren't hitting this. Added a commit that removes that TODO comment and passes in the same `opts.timeout` as the other calls.

Reply:
> the updates will automatically be installed if left unattended. Personally, I think that's fine, but I can see that being potentially undesirable. That said, I think it might be excessive to include yet another option flag for this particular case.

If there's no optional mods, it will do so anyway :P
I'm planning to rework this for #6, probably going to rewrite most of the UI to make it clearer.